src/components/vuestic-components/va-form/VaForm.vue

```diff
@@ -0,0 +1,162 @@
+<script>
+import { ContextPluginMixin, getContextPropValue } from '../../context-test/context-provide/ContextPlugin'
+
+const availableFormTags = ['VaInput', 'VaColorInput', 'VaDatePicker', 'VaCheckbox', 'VaRadio', 'VaSelect', 'VaFileUpload']
+const getAllChildren = (vm, children = []) => {
+  vm.$children.forEach(function (child) {
+    children.push(child)
+    child.$children.length > 0 && getAllChildren(child, children)
+  })
+  return children
+}
+
+export default {
+  name: 'VaForm',
+  mixins: [ContextPluginMixin],
+  props: {
+    lazyValidation: {
+      type: Boolean,
+      default () {
+        return getContextPropValue(this, 'lazyValidation', false)
+      },
+    },
+    autofocus: {
+      type: Boolean,
+      default () {
+        return getContextPropValue(this, 'autofocus', false)
+      },
+    },
+    tag: {
+      type: String,
+      default () {
+        return getContextPropValue(this, 'tag', 'div')
+      },
+    },
+  },
+  data () {
+    return {
+      valid: true,
+      hasResetButton: false,
+      hasSubmitButton: false,
+    }
+  },
+  mounted () {
+    this.$el.addEventListener('focusin', this.focus)
+    const formChild = getAllChildren(this)
```

Comment:
`focusable` should focus on first item.
Yet `this.$children` ordering has no relation whatsoever to visual ordering.

Reply:
Focusable components can be wrapped in another component.

Reply:
@Kreezag sorry, but this has no relation whatsoever to what I said.

Reply:
Fixed